Spack packages built with different specs have the same modulefile name

@chrisb13 reported the following:
https://github.com/ACCESS-NRI/ACCESS-OM2/pull/82#issuecomment-2434357382

@anton-seaice & @dougiesquire noticed that `module load` would pick the wrong `MOM5` Spack package. It looks like we need to make the modulefile names unique again.

Here's an example of a name collision:
```
$ find [...]/prerelease/apps/spack/0.22/release/modules/ -iname '2024.08.23'
[...]/prerelease/apps/spack/0.22/release/modules/linux-rocky8-x86_64/mom5/2024.08.23
[...]/prerelease/apps/spack/0.22/release/modules/linux-rocky8-x86_64_v4/mom5/2024.08.23
```

The default Spack configuration appends the Spack hash to the modulefile name.

Looks like there are changes that need to be made to build-cd to accommodate our preferred format:
```
Error: cice5: Version of dependency and projection do not match (2023.10.19 != 2023.10.19-{hash:7})
Error: mom5: Version of dependency and projection do not match (2024.08.23 != 2024.08.23-{hash:7})
Error: libaccessom2: Version of dependency and projection do not match (2023.10.26 != 2023.10.26-{hash:7})
Error: oasis3-mct: Version of dependency and projection do not match (2023.11.09 != 2023.11.09-{hash:7})
Error: Process completed with exit code 1.
```

Our preferred format:
```
          cice5: '{name}/2023.10.19-{hash:7}'
          mom5: '{name}/2024.08.23-{hash:7}'
          libaccessom2: '{name}/2023.10.26-{hash:7}'
          oasis3-mct: '{name}/2023.11.09-{hash:7}'
```

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -182,8 +182,11 @@ jobs:
               DEP_VER=$(yq ".spack.packages.\"$DEP\".require[0] | match(\"^@(?:git.)?([^=]*)\").captures[0].string" spack.yaml)
             fi
 
+            # Get the version from the module projection, for comparison with DEP_VER
+            # Projections are of the form '{name}/VERSION[-{hash:7}]', in which we only care about VERSION. For example, '{name}/2024.11.11', or '{name}/2024.11.11-{hash:7}'
             MODULE_NAME=$(yq ".spack.modules.default.tcl.projections.\"$DEP\"" spack.yaml)
-            MODULE_VER="${MODULE_NAME#*/}"  # Get 'version' from 'name/version' module, even if version contains '/'
+            MODULE_VER="${MODULE_NAME#*/}"  # Strip '{name}/' from '{name}/VERSION' module, even if VERSION contains '/'
+            MODULE_VER="${MODULE_VER%%-\{hash:7\}}"  # Strip a potential '-{hash:7}' appendix from the VERSION, since we won't have that in the DEP_VER
 
             if [[ "$DEP_VER" != "$MODULE_VER" ]]; then
               echo "::error::$DEP: Version of dependency and projection do not match ($DEP_VER != $MODULE_VER)"

